Feature/enhanced type support

WHenderson opened the pull request:
* Replaces visitor pattern / Extends type coverage
* Better support for complex nested types
* Attempts reference resolution directly before falling back to source maps
* Normalize path separators to support cross platform
* Replace usage of `rm` with `rimraf` in package scripts to support windows

nlepage commented on src/index.ts:
```diff
@@ -136,9 +136,9 @@ function findSymbolSourceFile(symbol: TSSymbol, project: ProjectReflection) {
 
     for (const source of sources) {
-      const srcFileName = path.resolve(srcDirPath, source)
+      const srcFileName = path.resolve(srcDirPath, source).replace(/\\/g, '/')
```
**suggestion:** Try to use [`path.normalize()`](https://nodejs.org/dist/latest-v16.x/docs/api/path.html#pathnormalizepath)?

WHenderson replied:
done

nlepage reviewed (commented):
Hi @WHenderson
Thanks for your contribution!
I don't have time to perform a full review right now.
I should be able to by the end of next week.

nlepage commented:
Hi @WHenderson 

Sorry for the delay.

I took the initiative of reordering the commits, in order to add the new examples before the visitor's refactoring.
I also added a commit with some tests on the new examples.

Turns out the new examples were already working without the refactoring (except for the Intersection but this was a bug in Typedoc itself, fixed here : https://github.com/TypeStrong/typedoc/pull/1910)

Ideally I'd like to have some examples that highlight the need for the refactoring.
Could you add some, maybe based on failing cases you would have on a project of yourself?

Thanks in advance.

WHenderson commented:
> Hi @WHenderson
> 
> Sorry for the delay.
> 
> I took the initiative of reordering the commits, in order to add the new examples before the visitor's refactoring. I also added a commit with some tests on the new examples.
> 
> Turns out the new examples were already working without the refactoring (except for the Intersection but this was a bug in Typedoc itself, fixed here : [TypeStrong/typedoc#1910](https://github.com/TypeStrong/typedoc/pull/1910))
> 
> Ideally I'd like to have some examples that highlight the need for the refactoring. Could you add some, maybe based on failing cases you would have on a project of yourself?
> 
> Thanks in advance.

A large number of the types weren't correctly linking up in my main mono-repo, so I debugged through the generation steps and came up with this fix. As you can tell from the order of commits, I didn't use TDD and am surprised that all the types correctly linked even without the changes.

If I find some time I'll see if I can create a test case that justifies the fix. Right now I'm just trying to get my repo to the stage I can make it public (sorting documentation and licensing).

Thanks for taking the time though and also for creating this extension in the first place!

nlepage commented:
Hi @WHenderson 
I published a new release [v0.2.0](https://github.com/nlepage/typedoc-plugin-resolve-crossmodule-references/releases/tag/v0.2.0) for `typedoc` v0.23 compatibility.
This new version doesn't rely on source maps anymore.
I picked up some of your commits on the main branch (windows compatibility, new examples and new tests).
Please let me know if the new version works for you.
Thanks.
